optimum/neuron/pipelines/transformers/base.py

```diff
@@ -0,0 +1,238 @@
+    if model is None:
+        model_id = SUPPORTED_TASKS[targeted_task]["default"]
+        if input_shapes is None:
+            input_shapes = {"batch_size": 1, "sequence_length": 128}
+            logger.warning(f"No input shapes provided, using default shapes, {input_shapes}")
+        model = SUPPORTED_TASKS[targeted_task]["class"][0].from_pretrained(model_id, export=True, **input_shapes)
+    # loads model from model id and converts it to neuronx optionally
+    elif isinstance(model, str):
+        model_id = model
+        neuronx_model_class = SUPPORTED_TASKS[targeted_task]["class"][0]
+        model = neuronx_model_class.from_pretrained(model, export=export, **model_kwargs, **input_shapes)
+    # uses neuron model
+    elif isinstance(model, NeuronBaseModel):
+        if tokenizer is None and load_tokenizer:
+            for preprocessor in model.preprocessors:
+                if isinstance(preprocessor, (PreTrainedTokenizer, PreTrainedTokenizerFast)):
+                    tokenizer = preprocessor
+                    break
+            if tokenizer is None:
+                raise ValueError(
+                    "Could not automatically find a tokenizer for the NeuronBaseModel, you must pass a tokenizer explictly"
+                )
+        if feature_extractor is None and load_feature_extractor:
+            for preprocessor in model.preprocessors:
+                if isinstance(preprocessor, SequenceFeatureExtractor):
+                    feature_extractor = preprocessor
+                    break
+            if feature_extractor is None:
+                raise ValueError(
+                    "Could not automatically find a feature extractor for the ORTModel, you must pass a "
+                    "feature_extractor explictly"
+                )
+        model_id = None
+    else:
+        raise ValueError(
+            f"""Model {model} is not supported. Please provide a valid model either as string or NeuroModel.
+            You can also provide non model then a default one will be used"""
+        )
+    return model, model_id, tokenizer, feature_extractor
+
+
+def pipeline(
+    task: str = None,
+    model: Optional[Any] = None,
+    tokenizer: Optional[Union[str, PreTrainedTokenizer]] = None,
+    feature_extractor: Optional[Union[str, PreTrainedFeatureExtractor]] = None,
+    use_fast: bool = True,
+    export: bool = False,
+    input_shapes: Optional[Dict[str, int]] = None,
+    use_auth_token: Optional[Union[str, bool]] = None,
+    revision: Optional[str] = None,
+    trust_remote_code: Optional[bool] = None,
+    *model_kwargs,
+    **kwargs,
+) -> Pipeline:
+    if task not in list(NEURONX_SUPPORTED_TASKS.keys()):
+        raise ValueError(
+            f"Task {task} is not supported for the optimum neuron pipeline. Supported tasks are {list(NEURONX_SUPPORTED_TASKS.keys())}"
+        )
+
+    if input_shapes is None and export:
+        input_shapes = {"batch_size": 1, "sequence_length": 128}
```

Comment:
To further support multiple choice and other vision tasks, we will need to infer the mandatory axes automatically. Or you want to hard code it for the moment to get the pipeline in first?

Reply:
lets keep it for now. We are not yet supporting those other tasks. We can implement defaults for each model in the corresponding "model class" later with warnings? 

It is important to get a first version out.

Reply:
`multiple-choice` is not a supported task atm.